src/main/java/me/gnat008/perworldinventory/data/MySQL.java

```diff
@@ -0,0 +1,288 @@
+
+            // Logout location
+            sql = "CREATE TABLE IF NOT EXISTS `" + prefix + "logout_location` (" +
+                    "`id` INT NOT NULL AUTO_INCREMENT," +
+                    "`pid` INT NOT NULL," +
+                    "`world` VARCHAR(255) NOT NULL," +
+                    "`x` DOUBLE NOT NULL," +
+                    "`y` DOUBLE NOT NULL," +
+                    "`z` DOUBLE NOT NULL," +
+                    "`pitch` FLOAT NOT NULL," +
+                    "`yaw` FLOAT NOT NULL," +
+                    "PRIMARY KEY (id)," +
+                    "FOREIGN KEY (pid) REFERENCES " + prefix + "players(pid));";
+            statement.executeUpdate(sql);
+
+            PwiLogger.info("MySQL tables created!");
+        }
+    }
+
+    private void setConnectionArguments() {
+        dataSource = new HikariDataSource();
+        dataSource.setPoolName("PerWorldInventorySQLPool");
+
+        dataSource.setMaximumPoolSize(poolSize);
+        dataSource.setJdbcUrl("jdbc:mysql://" + hostname + ":" + port + "/" + databaseName);
+
+        // Authentication
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+
+        // Encoding
+        dataSource.addDataSourceProperty("characterEncoding", "utf8");
+        dataSource.addDataSourceProperty("encoding","UTF-8");
+        dataSource.addDataSourceProperty("useUnicode", "true");
+
+        // Random stuff
+        dataSource.addDataSourceProperty("rewriteBatchedStatements", "true");
+        dataSource.addDataSourceProperty("jdbcCompliantTruncation", "false");
+
+        // Caching
+        dataSource.addDataSourceProperty("cachePrepStmts", "true");
+        dataSource.addDataSourceProperty("prepStmtCacheSize", "275");
+        dataSource.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+
+        PwiLogger.info("MySQL database parameters set!");
+    }
+
+    private void loadSettings(Settings settings) {
+        this.hostname = settings.getProperty(DatabaseProperties.HOSTNAME);
+        this.port = settings.getProperty(DatabaseProperties.PORT);
+        this.username = settings.getProperty(DatabaseProperties.USERNAME);
+        this.password = settings.getProperty(DatabaseProperties.PASSWORD);
+        this.databaseName = settings.getProperty(DatabaseProperties.DATABASE_NAME);
+        this.prefix = settings.getProperty(DatabaseProperties.PREFIX);
+        this.poolSize = settings.getProperty(DatabaseProperties.DATABASE_POOL_SIZE);
+
+        if (poolSize == -1) {
+            this.poolSize = Utils.getCoreCount();
+        }
+    }
+
+    @Override
+    public void close() {
+        if (dataSource != null && !dataSource.isClosed()) {
+            dataSource.close();
+        }
+    }
+
+    /**
+     * Log a SQLException to the console and logs.
+     *
+     * @param ex The raised exception.
+     */
+    private static void logSQLException(SQLException ex) {
+        PwiLogger.severe("Error during SQL operation:" + ex);
```

Comment:
for this and 2x below: unless you really want this, use `PwiLogger.severe("Error ...", ex)` instead